Index: b/include/EST_Chunk.h
===================================================================
--- a/include/EST_Chunk.h
+++ b/include/EST_Chunk.h
@@ -87,6 +87,8 @@
 
 #endif
 
+class EST_ChunkPtr;
+
  /************************************************************************/
  /*                                                                      */
  /* EST_Chunk is a use-counted chunk of memory. You shouldn't be able    */
@@ -100,6 +102,15 @@
  /*                                                                      */
  /************************************************************************/
 
+typedef  int EST_chunk_size;
+
+EST_ChunkPtr chunk_allocate(int bytes);
+EST_ChunkPtr chunk_allocate(int bytes, const char *initial, int initial_len);
+EST_ChunkPtr chunk_allocate(int bytes, const EST_ChunkPtr &initial, int initial_start, int initial_len);
+
+void make_updatable(EST_ChunkPtr &shared, EST_chunk_size inuse);
+void make_updatable(EST_ChunkPtr &shared);
+
 class EST_ChunkPtr;
 
 class EST_Chunk  {
