--- speech-tools-1.2.3.orig/include/EST_Val.h
+++ speech-tools-1.2.3/include/EST_Val.h
@@ -252,8 +252,10 @@
 
 inline const char *error_name(const EST_Val val) { return (EST_String)val;}
 
+
 // For consistency with other (user-defined) types in val
-inline EST_String string(const EST_Val &v) { return v.string(); }
+//
+inline EST_String String(const EST_Val &v) { return v.string(); }
 inline EST_Val est_val(const EST_String s) { return EST_Val(s); }
 inline EST_Val est_val(const char *s) { return EST_Val(s); }
 inline int Int(const EST_Val &v) { return v.Int(); }
