Index: b/sigpr/filter.cc
===================================================================
--- a/sigpr/filter.cc
+++ b/sigpr/filter.cc
@@ -427,7 +427,7 @@
 	
     // check frequency_response has dimension 2^N
     int N = fastlog2(frequency_response.n());
-    if(frequency_response.n() !=  (int)pow(2,(float)N)){
+    if(frequency_response.n() !=  (int)pow((float)2,(float)N)){
 	cerr << "Desired frequency response must have dimension 2^N" << endl;
 	return EST_FVector(0);
     }
@@ -478,10 +478,11 @@
     int i;
     int N=10;			// good minimum size
     
-    int fft_size = (int)pow(2, N);
+	// pow(float,int) ??
+    int fft_size = (int)pow((float)2, (float)N);
     while(fft_size < order*4){	// rule of thumb !?
 	N++;
-	fft_size = (int)pow(2, N);
+	fft_size = (int)pow((float)2, (float)N);
     }
     
     // freq response is from 0 to sampling freq and therefore
