--- speech-tools-1.2.3.orig/speech_class/EST_track_aux.cc
+++ speech-tools-1.2.3/speech_class/EST_track_aux.cc
@@ -351,7 +351,7 @@
     mean /= n;
     
     for (i = 0, var = 0.0; i < tr.num_frames(); ++i)
-	var += tr.track_break(i) ? 0.0 : pow(tr.a(i, channel) - mean, 2.0);
+	var += tr.track_break(i) ? 0.0 : pow(tr.a(i, channel) - mean, (float)2.0);
     
     var /= n;
     sd = sqrt(var);
@@ -369,7 +369,7 @@
     
     for (i = 0; i < size; ++i)
 	if (a.val(i) && b.val(i))
-	    sum += pow((a.a(i, channel) - b.a(i, channel)), 2.0);
+	    sum += pow((a.a(i, channel) - b.a(i, channel)), (float)2.0);
     
     sum = sqrt(sum / size);
     return sum;
@@ -461,7 +461,7 @@
     for (p = tl.head(); p; p = next(p))
 	for (i = 0; i < tl(p).num_frames(); ++i)
 	    if (!tl(p).track_break(i))
-		var +=  pow(tl(p).a(i, channel) - mean, 2.0);
+		var +=  pow(tl(p).a(i, channel) - mean, (float)2.0);
 
     var /= n;
     sd = sqrt(var);
