--- speech-tools-1.2.3.orig/speech_class/EST_wave_cuts.cc
+++ speech-tools-1.2.3/speech_class/EST_wave_cuts.cc
@@ -67,7 +67,7 @@
 	if (end < start)
 	    continue;
 	wave_subwave(a, sig, start, end-start);
-	filename = k->f("file");
+	filename = EST_String(k->f("file"));
 	a.set_name(filename + ext);
 	wl.append(a);
 	start = end;
@@ -87,7 +87,7 @@
     for (k = keylab.head(); k; k = next(k))
     {
 	end = k->F("end",0);
-	key_file_name = k->f("file");
+	key_file_name = EST_String(k->f("file"));
 	if (key_file_name == file)
 	{
 	    wave_subwave(part, sig, start, end-start);
