--- speech-tools-1.2.3.orig/speech_class/ssff.cc
+++ speech-tools-1.2.3/speech_class/ssff.cc
@@ -37,8 +37,8 @@
 /* used in emulabel                                                      */
 /*                                                                       */
 /*=======================================================================*/
-#include <fstream.h>
-#include <iostream.h>
+#include <fstream>
+#include <iostream>
 #include <stdlib.h>
 #include <math.h>
 #include <time.h>
@@ -99,7 +99,7 @@
     
     while (ts.peek() != "-----------------")
     {
-	c = ts.get();
+	c = EST_String(ts.get());
 	if (c == "Comment")
 	    ts.get_upto_eoln();
 	else if (c == "Start_Time")
@@ -124,9 +124,9 @@
 	}
 	else if (c == "Column")
 	{
-	    name = ts.get();
-	    type = ts.get();
-	    size = ts.get();
+	    name = EST_String(ts.get());
+	    type = EST_String(ts.get());
+	    size = EST_String(ts.get());
 	    cname = EST_String("Channel_")+itoString(num_channels);
 	    channels.set(cname+".name",name);
 	    channels.set(cname+".type",type);
@@ -141,7 +141,7 @@
 		 (c == "preemphasis") ||
 		 (c == "frame_duration"))
 	{
-		type = ts.get();
+		type = EST_String(ts.get());
 		if (type == "SHORT")
 		    tr.f_set(c,atoi(ts.get().string()));
 		else if (type == "DOUBLE")
