--- speech-tools-1.2.3.orig/stats/wagon/dlist.cc
+++ speech-tools-1.2.3/stats/wagon/dlist.cc
@@ -45,7 +45,7 @@
 /*  1987                                                                 */
 /*                                                                       */
 /*=======================================================================*/
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include "EST_Wagon.h"
 #include "EST_FMatrix.h"
@@ -83,7 +83,7 @@
 
     for (p=dataset.head(); p != 0; p=next(p))
     {
-	predict = dlist->predict(*dataset(p));
+	predict = EST_String(dlist->predict(*dataset(p)));
 	type = dataset.ftype(0);
 	real = wgn_discretes[type].name(dataset(p)->get_int_val(0));
 	pairs.add_item(real,predict,1);
